app/src/main/java/com/example/resoluteapp/InboxFragment.java

```diff
@@ -99,181 +81,209 @@ public void onDestroyView() {
+                            TextView tv2 = new TextView(requireContext());
+                            TextView tv3 = new TextView(requireContext());
+                            TextView tv4 = new TextView(requireContext());
+
+                            //set id for tablerow
+                            tr.generateViewId();
+
+                            //sets height of tablerow
+                            tr.setMinimumHeight(60);
+
+                            //sets the first text view with the username data for the exercise
+                            tv1.setText(username);
+                            tv1.setTextColor(Color.BLACK);
+                            tv1.setPadding(10, 10, 10, 10);
+                            tv1.setTextSize(12);
+                            tv1.setWidth(tl.getWidth() / 4);
+
+                            //adds it to the table row in the first column
+                            tr.addView(tv1, 0);
+
+                            //sets the second text view with the exercise data for the exercise
+                            tv2.setText(exercise);
+                            tv2.setTextColor(Color.BLACK);
+                            tv2.setPadding(10, 10, 10, 10);
+                            tv2.setTextSize(12);
+                            tv2.setWidth(tl.getWidth() / 4);
+
+                            //adds it to the table row in the second column
+                            tr.addView(tv2, 1);
+
+                            //sets the third text view with the amount data for the exercise
+                            tv3.setText(amount);
+                            tv3.setTextColor(Color.BLACK);
+                            tv3.setPadding(10, 10, 10, 10);
+                            tv3.setTextSize(12);
+                            tv3.setWidth(tl.getWidth() / 4);
+
+                            //adds it to the table row in the third column
+                            tr.addView(tv3, 2);
+
+
+                            //sets the fourth text view with the unit data for the exercise
+                            tv4.setText(unit);
+                            tv4.setTextColor(Color.BLACK);
+                            tv4.setPadding(10, 10, 10, 10);
+                            tv4.setTextSize(12);
+                            tv4.setWidth(tl.getWidth() / 4);
+
+                            //adds it to the table row in the fourth column
+                            tr.addView(tv4, 3);
+
+                            // Set a click listener on the TableRow to trigger the reply dialog
+                            tr.setOnClickListener(v -> showReplyDialog(username, docId)); // Pass docId
+
+                            //adds the row to the table
+                            tl.addView(tr);
                         }
                     }
                 });
     }
-}
\ No newline at end of file
+
+    // Function to refresh the table after replying or removing a row
+    private void refreshTable() {
```

Comment:
This function seems unnecessary, since the navigation path
NavHostFragment.findNavController(InboxFragment.this).navigate(R.id.action_inboxFragment_self);
already updates the table by reloading the fragment. I see you overhauled the way the ReplyDialog is shown to the user, so if this change is deemed necessary then it's okay. Otherwise, it almost feels like a bunch of unnecessary time spent working.